src/components/navbar/navDropdown/NavDropdown.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Link } from 'react-router-dom';
+import classes from './NavDropdown.module.scss';
+import PropTypes from 'prop-types';
+import NavLinkSection from '../navLinkSection/NavLinkSection';
+
+const NavDropdown = ({
+    display,
+    navLink,
+    logo,
+    closeDropdownNav,
+    toggleNewsModal,
+}) => {
+    const initTransitionStyle = {
+        transform: 'translate3d(0, -110%, 0)',
+        visibility: 'hidden',
+    };
+    const terminalTransitionStyle = {
+        transform: 'translate3d(0, 0, 0)',
+        visibility: 'visible',
+    };
+    const [transitionStyle, setTransitionStyle] = useState(initTransitionStyle);
+    const ref = useRef();
+
+    useEffect(() => {
+        // mounting animation
+        if (display) {
+            setTransitionStyle((prevState) => {
+                return {
+                    ...prevState,
+                    ...terminalTransitionStyle,
+                };
+            });
+        } else {
+            setTransitionStyle((prevState) => {
+                return {
+                    ...prevState,
+                    ...initTransitionStyle,
+                };
+            });
+        }
```

Comment:
You can use the spread operator conditionally as so:

```
useEffect(() => {
          // mounting animation
          setTransitionStyle((prevState) => {
                return {
                    ...prevState,
                    ...(display ? terminalTransitionStyle : initTransitionStyle),
                };
            });
}
```

Reply:
Ohh very nice, I did not think it could be written this way, thank you for sharing.